### PR TITLE
`grid` and limited `Filter` caching

### LINKED ISSUE #37: Filter object could benefit from a cached return value of evaluate_grid
There's likely multiple times that an unchanged `Filter` object is asked to do a `evaluate_grid` on the same input resolution (investigate if/where this is happening). If so, the `Filter` object could serve up the same pre-evaluated grid of values when asked for.

### PR DESCRIPTION
**Pull request by garrettwrong**
This PR is considering some basic caching that is available using the standard Python library in 3.9+.  It should compliment the more sophisticated caching we do in (Basis etc) classes.

The general scheme employed here is i) to allow caching control via our configuration ii) to employ caching in a way that has minimal impact to memory footprint.

Closes #37

### CODE REVIEW

**Review by j-c-c — APPROVED** [review]
LGTM!

File: src/aspire/utils/misc.py [hunk 1]
```diff
@@ -170,15 +170,15 @@ def gaussian_2d(size, mu=(0, 0), sigma=(1, 1), indexing="yx", dtype=np.float64):
     # Construct centered mesh
     g = grid_2d(size, shifted=False, normalized=False, indexing=indexing, dtype=dtype)
 
+    X, Y = g["x"], g["y"]
```
Comment (janden) [inline src/aspire/utils/misc.py:173]: Is this just cosmetic or is there a deeper reason here?

**Review by janden — APPROVED** [review]
Looks great! Just a small question.

File: src/aspire/utils/misc.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/misc.py:173]: Avoids mutating the dictionary performing the swap.

**Pull request merged**